src/firefox/index.js

```diff
@@ -249,9 +281,45 @@ export async function useProfile(
     app,
     configureThisProfile = configureProfile,
     customPrefs = {},
+    createProfileFinder = defaultCreateProfileFinder,
   }: UseProfileParams = {},
 ): Promise<FirefoxProfile> {
-  const profile = new FirefoxProfile({destinationDirectory: profilePath});
+  let profile;
+  let defaultProfilePath = '';
+  let defaultDevProfilePath = '';
+  const finder = createProfileFinder();
+  try {
+    const dirExists = await isDirectory(profilePath);
+    if (await finder.hasProfileName('default')) {
+      defaultProfilePath = await finder.getPath('default');
```

Comment:
As an alternative approach to this two `if (await finder.hasProfileName(...)) { ... = await finder.getPath(...); }` we can opt to return from `createProfileFinder` a single function that returns a string (the profile path) if there is such a profile name and `undefined` (or `null`) otherwise 
(in other words: we can merge the `hasProfileName` and `getPath` function defined at line 253 into a single helper function with the behavior described above).

This change can make this method implementation cleaner and simpler, turning it into something like:

```js
const getProfilePath = createProfileFinder();
const dirExists = await isDirectory(profilePath);

let destinationDirectory;

if (dirExists) {
  ...
  if (profilePath === getProfilePath('default') ||
      profilePath === getProfilePath('dev-edition-default')) {
    throw new WebExtError(...);
  }
  
  destinationDirectory = profilePath;
} else {
  ...
  if (profilePath === 'default' ||
      profilePath === 'dev-edition-default') {
    throw new WebExtError(...);
  }

  destinationDirectory = getProfilePath(profilePath);

  if (!destinationDirectory) {
    throw new UserError(
      `The request "${profilePath}" profile name cannot be resolved to a profile path`
    );
  }
}

....

const profile = new FirefoxProfile({destinationDirectory});
return await configureThisProfile(profile, {app, customPrefs});
```